test/e2e/config/e2e_conf.yaml

```diff
@@ -18,6 +18,15 @@ providers:
 - name: cluster-api
   type: CoreProvider
   versions:
+  - name: v0.4.7
+    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.4.7/core-components.yaml"
+    type: "url"
+    contract: v1alpha4
+    replacements:
+      - old: --metrics-addr=127.0.0.1:8080
+        new: --metrics-addr=:8080
+    files:
+      - sourcePath: "../data/shared/v1beta1/metadata.yaml"
```

Comment:
I am not sure I like the idea of having single metadata regardless of an API type, v1beta1 metadata.yaml for the release that is from v1a4 API makes it confusing

Reply:
Would it be ok if we just skip the `v1beta1` in the path? So we would have `data/shared/metadata.yaml`.
I mean it just lists all the releases and what contract to use, right? So why would we need to have them separate for v1alpha4 and v1beta1?

Reply:
@lentzi90 actually, why should that file be the same and reused for CAPI and CAPM3? 
CAPIs' release series was always mapped 1:1, i.e:

```
releaseSeries:
  - major: 1
    minor: 1
    contract: v1beta1
  - major: 1
    minor: 0
    contract: v1beta1
  - major: 0
    minor: 4
    contract: v1alpha4
  - major: 0
    minor: 3
    contract: v1alpha3
```

whereas CAPM3 had 2 minor releases for the same contract, i.e:


```
releaseSeries:
- major: 1
  minor: 0
  contract: v1beta1
- major: 0
  minor: 5
  contract: v1alpha4
- major: 0
  minor: 4
  contract: v1alpha3
- major: 0
  minor: 3
  contract: v1alpha3
```

So, it means in this PR, we have this metadata:

```
apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
kind: Metadata
releaseSeries:
  - major: 0
    minor: 4
    contract: v1alpha3
  - major: 0
    minor: 5
    contract: v1alpha4
  - major: 1
    minor: 0
    contract: v1beta1
```

being used for CAPI release v0.4.7. 

Moreover, as releases keep coming up, in my opinion, it is not convenient to use the same metadata.yaml for each e2e test and that is the practice I have seen in other CAPI provider repos, for ref. you can check here:

- https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/main/test/e2e/data/shared
- https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/test/e2e/data/shared

Reply:
I have followed the same pattern in this PR I still think moving to a single metadata (if agreed) can be done independently of the upgrade changes

Reply:
I didn't mean we would use the same for CAPI and CAPM3, I mean I don't understand why we need to have one for v1beta1 that has all the older versions in it and then also have one for v1alpha5 and v1alpha4? But I see the other providers do this so I guess there must be a good reason.